src/main/java/attendance/service/AttendanceService.java

```diff
@@ -0,0 +1,117 @@
+import attendance.domain.Attendances;
+import attendance.domain.PenaltyCrew;
+import attendance.dto.AttendanceInfoDto;
+import attendance.dto.CrewAttendanceDto;
+import attendance.dto.EditResponseDto;
+import attendance.dto.FileRequestDto;
+import attendance.dto.PenaltyCrewDto;
+import attendance.utils.AttendanceReader;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AttendanceService {
+
+    private Attendances attendances;
+    private final AttendanceReader attendanceReader;
+
+    public AttendanceService(AttendanceReader attendanceReader) {
+        this.attendanceReader = attendanceReader;
+    }
+
+    public void init() {
+        List<FileRequestDto> dtos = attendanceReader.read();
+        List<Attendance> convertedAttendances = dtos.stream()
+                .map(dto -> new Attendance(dto.name(), dto.date(), dto.time()))
+                .toList();
+        attendances = new Attendances(convertedAttendances);
+    }
+
+    public void findName(String name) {
+        attendances.checkName(name);
+    }
+
+    public void checkByNameAndDate(String name, LocalDate today) {
+        attendances.findLocalTimeByNameAndDate(name, today);
+    }
+
+    public void insertAttendance(String name, LocalDate today, LocalTime time) {
+        checkByNameAndDate(name, today);
+        Attendance attendance = new Attendance(name, today, time);
+        this.attendances = attendances.add(attendance);
+    }
+
+    public String getAttendanceStatus(LocalDate date, LocalTime time) {
+        return AttendanceStatus.of(date, time).getKorean();
+    }
+
+    public EditResponseDto edit(String name, LocalDate date, LocalTime editTime) {
+        LocalTime oldTime = editAttendance(name, date, editTime);
+        String oldStatus = getAttendanceStatus(date, oldTime);
+        String editStatus = getAttendanceStatus(date, editTime);
+
+        return new EditResponseDto(date, oldTime, editTime, oldStatus, editStatus);
+    }
+
+    public CrewAttendanceDto getCrewAttendance(String name, LocalDate today) {
+        Map<LocalDate, AttendanceInfoDto> attendanceInfos = getAttendanceInfos(name, today);
+        List<Integer> counts = getAttendanceCounts(name, today);
+        AttendancePenalty penalty = AttendancePenalty.find(counts);
+        return CrewAttendanceDto.of(name, attendanceInfos, counts, penalty, today);
+    }
+
+    private Map<LocalDate, AttendanceInfoDto> getAttendanceInfos(String name, LocalDate today) {
+        Map<LocalDate, AttendanceInfoDto> map = new HashMap<>();
+        List<Attendance> attendanceList = attendances.findByNameAndDateWithAscend(name, today);
+
+        for (Attendance attendance : attendanceList) {
+            AttendanceInfoDto dto = AttendanceInfoDto.toDto(attendance);
+            map.put(dto.attendanceDate(), dto);
+        }
+
+        return map;
```

Comment:
1. Map 객체를 `attendances.findByNameAndDateWithAscend(name, today);` 보다 먼저 생성할 필요는 없을 것 같아요~
만약 메서드 호출을 하다가 실패하면 미리 생성해둔 Map 객체는 의미가 없어지니까요!

```suggestion
        List<Attendance> attendanceList = attendances.findByNameAndDateWithAscend(name, today);

        Map<LocalDate, AttendanceInfoDto> map = new HashMap<>();
        for (Attendance attendance : attendanceList) {
            AttendanceInfoDto dto = AttendanceInfoDto.toDto(attendance);
            map.put(dto.attendanceDate(), dto);
        }
        return map;
```

2. 여기도 map 이라는 변수명을 명확하게 변경해보면 좋겠어요!

Reply:
객체를 생성하는 것과 초기화 하는 로직은 최대한 붙이는 것이 좋다는걸 배웠습니다. 필요한 객체를 생성하기 이전에 로직 호출이 중단될 경우 쓸모없는 객체가 생성되는 것을 막을수 있으니까요